react/src/lib/components/DeckGLMap/components/ContinuousLegend.tsx

```diff
@@ -1,106 +1,96 @@
+        const linearGradient = defs
+            .append("linearGradient")
+            .attr("id", "linear-gradient")
+            .attr("x1", "0%")
+            .attr("x2", "100%") //since it's a horizontal linear gradient
+            .attr("y1", "0%")
+            .attr("y2", "0%");
+        // append multiple color stops by using D3's data/enter step
+        linearGradient
+            .selectAll("stop")
+            .data(itemColor)
+            .enter()
+            .append("stop")
+            .attr("offset", function (data) {
+                return data.offset + "%";
+            })
+            .attr("stop-color", function (data) {
+                return data.color;
+            });
 
-        const canvas = select(selected_id)
-            .style("width", 150 + "px")
-            .append("canvas")
-            .attr("height", legendheight + 5 - margin.top - margin.bottom)
-            .attr("width", 1)
-            .style(
-                "height",
-                legendheight + 5 - margin.top - margin.bottom + "px"
-            )
-            .style(
-                "width",
-                legendwidth + 13 - margin.left - margin.right + "px"
-            )
-            .style("border", "1px solid")
-            .node();
-
-        if (canvas) {
-            const context = canvas.getContext("2d");
-            const legendscale = scaleLinear()
-                .range([legendheight - margin.top - margin.bottom, 0])
-                .domain(colorscale.domain());
+        // append title
+        svgLegend
+            .append("text")
+            .attr("class", "legendTitle")
+            .attr("x", 25)
+            .attr("y", 20)
+            .style("text-anchor", "left")
+            .text(dataObjectName);
 
-            if (context) {
-                const image = context.createImageData(1, legendheight);
-                range(legendheight).forEach(function (i) {
-                    const c = rgb(colorscale(legendscale.invert(i)));
-                    image.data[4 * i] = c.r;
-                    image.data[4 * i + 1] = c.g;
-                    image.data[4 * i + 2] = c.b;
-                    image.data[4 * i + 3] = 255;
-                });
-                context.putImageData(image, 0, 0);
-            }
+        // draw the rectangle and fill with gradient
+        svgLegend
+            .append("rect")
+            .attr("x", 25)
+            .attr("y", 30)
+            .attr("width", 250)
+            .attr("height", 25)
+            .style("fill", "url(#linear-gradient)");
 
-            const legendaxis = axisRight(legendscale)
-                .scale(legendscale)
-                .tickValues(legendscale.domain());
-            const svg = select(selected_id)
-                .append("svg")
-                .attr("height", legendheight - 3 + "px")
-                .attr("width", legendwidth - 20 + "px");
+        //create tick marks
+        const xLeg = scaleLinear().domain([min, max]).range([10, 258]);
```

Comment:
Why are we using hard coded range?

Reply:
Here range is used to adjust the position of tick line according to legend, which will always be same, so I feel that's fine to give hard coded range.